pkg/auth/dependency/principal/oauth/error.go

```diff
@@ -0,0 +1,7 @@
+package oauth
+
+import (
+	"github.com/skygeario/skygear-server/pkg/core/skyerr"
+)
+
+var ErrIdentityNotFound = skyerr.NotFound.WithReason("OAuthIdentityNotFound").New("oauth identity not found")
```

Comment:
I think this error can be moved up to `principal` package, or even update the existing principal not found error, since there is no need for client to know the type of identity.

Reply:
I updated the principal not found error to skyerr. However, we still have some error in `principal/password/error.go` that may overlap with those in `principal/error.go`

Reply:
Oh I forgot about the error in password package. If so, I think these skyerrs should belongs to the handler layer instead. So providers would return a simple error (`ErrNotFound`), and handlers would translate it into skyerrs for each group of APIs.

Reply:
So revert to my original change?

Reply:
Yes, sorry for this. It would be better to also move the not found errors into the handler (i.e. into `loginid` and `sso` packages in handlers)

Reply:
Updated.